operator/src/main/java/io/strimzi/kafka/access/KafkaAccessReconciler.java

```diff
@@ -95,101 +97,103 @@ public UpdateControl<KafkaAccess> reconcile(final KafkaAccess kafkaAccess, final
-        if (kafkaAccessSecretEventSource == null) {
-            throw new IllegalStateException("Event source for Kafka Access Secret not initialized, cannot reconcile");
-        }
-        kafkaAccessSecretEventSource.get(new ResourceID(secretName, kafkaAccessNamespace))
-                .ifPresentOrElse(secret -> {
-                    final Map<String, String> currentData = secret.getData();
-                    if (!data.equals(currentData)) {
-                        kubernetesClient.secrets()
-                                .inNamespace(kafkaAccessNamespace)
-                                .withName(secretName)
-                                .edit(s -> new SecretBuilder(s).withData(data).build());
-                    }
-                }, () -> kubernetesClient
-                        .secrets()
+
+        // Use the Context to get the Secret from the cache. Secret is secondary resource, KafkaAccess is primary resource.
+        Optional<Secret> existingSecret = context.getSecondaryResource(Secret.class, KAFKA_ACCESS_SECRET_EVENT_SOURCE);
+
+        if (existingSecret.isPresent()) {
+            final Map<String, String> currentData = existingSecret.get().getData();
+            if (!data.equals(currentData)) {
+                LOGGER.info("Updating existing secret {}/{}", kafkaAccessNamespace, secretName);
+                kubernetesClient.secrets()
                         .inNamespace(kafkaAccessNamespace)
-                        .resource(
-                                new SecretBuilder()
-                                        .withType(SECRET_TYPE)
-                                        .withNewMetadata()
-                                        .withName(secretName)
-                                        .withLabels(commonSecretLabels)
-                                        .withOwnerReferences(
-                                                new OwnerReferenceBuilder()
-                                                        .withApiVersion(kafkaAccess.getApiVersion())
-                                                        .withKind(kafkaAccess.getKind())
-                                                        .withName(kafkaAccessName)
-                                                        .withUid(kafkaAccess.getMetadata().getUid())
-                                                        .withBlockOwnerDeletion(false)
-                                                        .withController(false)
-                                                        .build()
-                                        )
-                                        .endMetadata()
-                                        .withData(data)
-                                        .build()
-                        )
-                        .create()
-            );
+                        .withName(secretName)
+                        .edit(s -> new SecretBuilder(s).withData(data).build());
```

Comment:
@im-konge do you think we should change this to patch and use the same logic as in Strimzi with trying without force and then with force? As far as I can see edit clones the resource then calls patch anyway, so I think we can use patch directly regardless of whether we choose to use force.

Reply:
@im-konge Can you point me to the logic in Strimzi please?

Reply:
Yeah I think we do the same as what you linked there, patch without force but handle the error and patch with force if needed